tidy3d/plugins/adjoint/components/simulation.py

```diff
@@ -127,13 +128,12 @@ def _no_overlap(cls, val):
             for j, in_struct_bck in enumerate(in_structs_background):
 
                 # if the contracted geometry intersects with a background structure, raise (overlap)
-                if in_geometry.intersects(in_struct_bck.geometry) and not warning_triggered:
+                if in_geometry.intersects(in_struct_bck.geometry):
```

Comment:
I'm not sure if the idea here was for this warning to be triggered for all structures now. However, note that all the warnings I get in the slab notebook if I set `spc=0` still always say spacing between structure 0 and 1, never between 1 and 2 or 2 and 3. I guess have a look into this when you update the notebook as we discussed.

Reply:
seems fixed on my end, as discussed